crates/driver/src/main.rs

```diff
@@ -0,0 +1,55 @@
+use clap::Parser;
+use driver::api::serve_api;
+use std::time::Duration;
+
+#[tokio::main]
+async fn main() {
+    let args = driver::arguments::Arguments::parse();
+    shared::tracing::initialize(args.log_filter.as_str(), args.log_stderr_threshold);
+    tracing::info!("running driver with validated arguments:\n{}", args);
+
+    let (shutdown_sender, shutdown_receiver) = tokio::sync::oneshot::channel();
+    let serve_api = serve_api(args.bind_address, async {
+        let _ = shutdown_receiver.await;
+    });
+
+    futures::pin_mut!(serve_api);
+    tokio::select! {
+        result = &mut serve_api => tracing::error!(?result, "API task exited"),
+        _ = shutdown_signal() => {
+            tracing::info!("Gracefully shutting down API");
+            shutdown_sender.send(()).expect("failed to send shutdown signal");
+            match tokio::time::timeout(Duration::from_secs(10), serve_api).await {
+                Ok(inner) => inner.expect("API failed during shutdown"),
+                Err(_) => tracing::error!("API shutdown exceeded timeout"),
+            }
+        }
+    };
+}
+
+#[cfg(unix)]
+async fn shutdown_signal() {
```

Comment:
I'm not sure if the signal stuff is needed. I think we added that to the api so that when the pod gets restarted it would complete the in flight requests instead of just dropping them. For this pod that doesn't seem important. On the other hand doesn't hurt to have it either.